src/main/java/build/buildfarm/metrics/prometheus/PrometheusPublisher.java

```diff
@@ -86,79 +37,9 @@ public static void startHttpServer(int port) {
     }
   }
 
-  public static void startHttpServer(int port, int exeStages, int ifStages) {
-    numExecuteStages = exeStages;
-    numInputFetchStages = ifStages;
-    startHttpServer(port);
-  }
-
   public static void stopHttpServer() {
     if (server != null) {
       server.stop();
     }
   }
-
-  public static void updateCpuQueueSize(long val) {
-    cpuQueueSize.set(val);
-  }
-
-  public static void updateGpuQueueSize(long val) {
-    gpuQueueSize.set(val);
-  }
-
-  public static void updatePreQueueSize(long val) {
-    preQueueSize.set(val);
-  }
-
-  public static void updateDispatchedOperationsSize(long val) {
-    dispatchedOperations.set(val);
-  }
-
-  public static void updateWorkerPoolSize(int val) {
-    workerPoolSize.set(val);
-  }
-
-  public static void updateExecutionSuccess() {
-    executionSuccess.inc();
-  }
-
-  public static void updateCompletedOperations() {
-    completedOperations.inc();
-  }
-
-  public static void updateActionResults(long val) {
-    actionResults.observe(val);
-  }
-
-  public static void updateMissingBlobs(long val) {
-    missingBlobs.observe(val);
-  }
-
-  public static void updateExecutionTime(double val) {
-    executionTime.observe(val);
-  }
-
-  public static void updateExecutionStallTime(double val) {
-    executionStallTime.observe(val);
-  }
-
-  public static void updateExecutionSlotUsage(int val) {
-    executionSlotUsage.set(val);
-  }
-
-  public static void updateInputFetchTime(double val) {
-    inputFetchTime.observe(val);
-  }
-
-  public static void updateInputFetchStallTime(double val) {
-    inputFetchStallTime.observe(val);
-  }
-
-  public static void updateInputFetchSlotUsage(int val) {
-    inputFetchSlotUsage.set(val);
-  }
-
-  public static void updateClusterUtilization() {
-    clusterUtilization.set(executionSlotUsage.get() * 100 / numExecuteStages);
-  }
```

Comment:
Is the diff removing any metrics? I see we're pulling metric gathering out of the backplane and into the instance. But I'm not sure if we've shrunk the number of metrics because of it.

Reply:
I realized that with having execute stage metric our cluster utilization calculation is very simple now and can just be calculated on the dashboard side.